src/util/io.c

```diff
@@ -59,6 +59,10 @@ read_all (
                          fd, errno, strerror (errno));
             return recvd_total;
```

Comment:
Not related to this patch, but `recvd` be returned instead to propagate the error code to the caller?

Reply:
not related, we can take this up in another issue / PR